### PR TITLE
[Display] Add Drawer

### PR DESCRIPTION
**Pull request by Balkeo**

### CODE REVIEW

**Comment by MatouTechnologies** [comment]
This pull request has been deployed to Vercel.
<table>
  <tr>
    <td><strong>✅ Preview:</strong></td>
    <td><a href='https://matou-technologies-purrfect-ui-storybook-66.vercel.app'>https://matou-technologies-purrfect-ui-storybook-66.vercel.app</a></td>
  </tr>
  <tr>
    <td><strong>🔍 Inspect:</strong></td>
    <td><a href='https://vercel.com/matou-technologies/purrfect-ui/6YWiTtRDEC2iZeSByQChpB2EfnKt'>https://vercel.com/matou-technologies/purrfect-ui/6YWiTtRDEC2iZeSByQChpB2EfnKt</a></td>
  </tr>
</table>


<hidden purpose="for-rewritable-pr-comment-action-use" value="deploy-preview-storybook"></hidden>

**Review by srobfr — APPROVED** [review]

File: src/display/Drawer/Drawer.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  it("Should not show the drawer", async () => {
+    render(
+      <Drawer isOpen={false} onClose={vi.fn()}>
+        <div>Content</div>
+      </Drawer>,
+    );
+
+    const content = screen.queryByText(/content/i);
+
+    expect(content).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (Balkeo) [reply src/display/Drawer/Drawer.spec.tsx:32]: The query* function has to be used when we are not sure the element is in the document

File: src/display/Drawer/Drawer.spec.tsx [hunk 2]
```diff
@@ -45,4 +51,95 @@ describe("Drawer", () => {
...
+    );
+
+    await waitFor(() => screen.getByText(/content/i));
+
+    const modal = screen.getByTestId("side-sheet");
+
+    expect(modal).toBeInTheDocument();
+  });
+
+  it("Should show the bottom sheet", () => {
```
Comment (MarcoPOtard) [inline src/display/Drawer/Drawer.spec.tsx:77]: In this test, you check if the side sheet is not in the document, but you don't test if the bottom sheet is present no ?

File: src/display/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { useIsMdScreen } from "src/guidelines/theme/mediaQueries";
 export type DrawerProps = {
   openFrom?: "left" | "right";
   isOpen: boolean;
-  header?: React.ReactNode;
+  header: React.ReactNode;
```
Comment (MarcoPOtard) [inline src/display/Drawer/Drawer.tsx:9]: Why did you delete the nullable to the header ? the header is still nullable, right ?

File: src/display/Drawer/SideSheet.tsx [hunk 4]
```diff
@@ -109,10 +109,17 @@ const StyledDialog = styled(Dialog)`
...
+  padding: 0;
+  font-weight: normal;
+  font-size: 1rem;
+`;
+
 type SideSheetProps = {
   isOpen: boolean;
   openFrom: "left" | "right";
-  header?: React.ReactNode;
+  header: React.ReactNode;
```
Comment (MarcoPOtard) [inline src/display/Drawer/SideSheet.tsx:122]: same question
Why did you delete the nullable to the header ? the header is still nullable, right ?
In the stories, you didn't put header on props.
